docs/src/rules/no-extra-parens.md

```diff
@@ -21,6 +21,8 @@ This rule always ignores extra parentheses around the following:
 * immediately-invoked function expressions (also known as IIFEs) such as `var x = (function () {})();` and `var x = (function () {}());` to avoid conflicts with the [wrap-iife](wrap-iife) rule
 * arrow function arguments to avoid conflicts with the [arrow-parens](arrow-parens) rule
 
+Problems reported by this rule can be fixed automatically, except when removing the parentheses would create a new directive.
```

Comment:
I was thinking more that we need to show some examples, as I don't know that most people will understand what "removing the parentheses would create a new directive" actually means.

Reply:
Thanks for the feedback @nzakas! Would it be also fine for you if we disable the autofix for all string literals that are direct children of an `ExpressionStatement`, as suggested [in this comment](https://github.com/eslint/eslint/pull/17022#discussion_r1152293435) by @mdjermanovic? This would improve interoperability with third-party rules at the expenses of a few autofixes in some rare edge cases.